### DISCUSSION TITLE
How to fit images in qCarousel?

### DISCUSSION THREAD

**Discussion by DrakoPOD** [quasarframework/quasar#9437] [General - Components / Directives / etc]
I have a Carousel, every image has different size, some fit correctly, other are crops and have problems on windows resize. Images come from an axios request. Also, the Carousel must have thumbnails.

**Comment by ontwikkelfabriek** [comment]
@DrakoPOD 
> I have a Carousel, every image has different size, some fit correctly, other are crops

What would you like to happen/see in the Carousel? You have different sized images ( and different aspect ratio I guess) and you don't want crops, and I guess you don't want the Carousel to change size all the time....

For better support please create a simple codepen.io( very simple) to demonstrate your problem:
https://codepen.io/rstoenescu/pen/VgQbdx

**Reply by DrakoPOD** [reply]
https://codepen.io/drakopod/pen/gOmGvxg

**Reply by ontwikkelfabriek** [reply]
@DrakoPOD 

How about this I created a carousel that have a `q-img` with the `contain` property in the default qslide slot :
https://quasar.dev/vue-components/img#contain-mode

https://codepen.io/ontwikkelfabriek/pen/jOBGjdN

 Downside is that the Carousels thumbnails don't auto generate anymore, you could probably fix that manually...

**Reply by DrakoPOD** [reply]
Thanks @ontwikkelfabriek, I already know that, I'm just trying to find a "legit" solution, in case that I had missed something in the docs, I like that my code looks clean as possible, and don't make unnecessary code for performance. 

You can also use `v-slot:navigation-icon` for your example ([docs](https://next.quasar.dev/vue-components/carousel#custom-navigation)):

```vue
<template v-slot:navigation-icon="{ active, onClick, name }">
  <q-btn v-if="active" color="transparent" flat dense @click="onClick" style="height: 60px; width: 100px; opacity: 1">
    <q-img :src="img[name]"/>
  </q-btn>
  <q-btn v-else color="transparent" flat dense @click="onClick" style="height: 60px; width: 100px; opacity: 0.5">
     <q-img :src="img[name]"/>
  </q-btn>
</template>
```
